pkg/cloudprovider/gce/gce.go

```diff
@@ -379,11 +383,53 @@ func (gce *GCECloud) CreateTCPLoadBalancer(name, region string, externalIP net.I
 		return nil, err
 	}
 
+	allowedPorts := make([]string, len(ports))
+	for ix := range ports {
+		allowedPorts[ix] = strconv.Itoa(ports[ix].Port)
+	}
+
+	hostTag := gce.computeHostTag(hosts[0])
+
+	firewall := &compute.Firewall{
+		Name:         makeFirewallName(name),
+		Description:  fmt.Sprintf("KubernetesAutoGenerated_OnlyAllowTrafficForDestinationIP_%s", fwd.IPAddress),
+		Network:      gce.gceNetworkName(),
+		SourceRanges: []string{"0.0.0.0/0"},
+		TargetTags:   []string{hostTag},
+		Allowed: []*compute.FirewallAllowed{
+			{
+				IPProtocol: "tcp",
+				Ports:      allowedPorts,
+			},
+		},
+	}
+	if op, err = gce.service.Firewalls.Insert(gce.projectID, firewall).Do(); err != nil {
+		return nil, err
+	}
+	if op.Status == "DONE" && op.Error != nil {
+		// An error occurred
+		if op.HttpErrorStatusCode == http.StatusConflict {
+			glog.V(4).Infof("Detected an already existing firewall: %s", firewall.Name)
+		} else {
+			var msg string
+			if len(op.Error.Errors) > 0 {
+				msg = op.Error.Errors[0].Message
+			}
+			return nil, fmt.Errorf("GCE Operation error: %s (%d %s)", msg, op.HttpErrorStatusCode, op.HttpErrorMessage)
```

Comment:
You should be able to rely on the logic in waitForOp to do this error handling work for you, as it just returns an error. That'd make this much more consistent with the other errors returned from these functions